### PR TITLE
Add entity financial account bridge entity

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Add entity financial account bridge entity

Before
----------------------------------------
Bridge entity not described

After
----------------------------------------
Now it is

Technical Details
----------------------------------------
@colemanw I've been digging around & can see that this is required -but is not in itself enough - ie 

image

What else is needed?

Comments
----------------------------------------

### CODE REVIEW

**Comment by colemanw** [comment]
@eileenmcnaughton if the table doesn't contain exactly 2 FKs, then you have to add an annotation to say which 2 entities it is supposed to link together. Like this:
https://github.com/civicrm/civicrm-core/blob/9f338c098614d0316a48629cae34986bd1a36e02/Civi/Api4/RelationshipCache.php#L26

**Comment by eileenmcnaughton** [comment]
@colemanw ah - so there is an entity_id -> entity_table

In addition I think the account_relationship option values would work rather like the record_type_id in activity contact table (ie they are almost like different entitiies)

image

File: Civi/Api4/EntityFinancialAccount.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace Civi\Api4;
+
+/**
+ * EntityFinancialAccount. Joins financial accounts to financial types.
+ *
+ * @see https://docs.civicrm.org/dev/en/latest/financial/financialentities/#financial-accounts
+ *
+ * @bridge entity_id financial_account_id
+ *
```
Comment (colemanw) [inline Civi/Api4/EntityFinancialAccount.php:27]: @eileenmcnaughton if you want the AccountRelationship field automatically shown during join (like relationship type id is shown for related contacts or activity contact type is shown for activities) do this:
```suggestion
 *  @ui_join_filters account_relationship
```

Comment (eileenmcnaughton) [reply Civi/Api4/EntityFinancialAccount.php:28]: @colemanw I've pushed that up - although note the whole thing doesn't work atm

**Comment by colemanw** [comment]
I see the problem @eileenmcnaughton. The field `entity_table` doesn't have any pseudoconstant, therefore the joiner has no idea what possible tables it can join to.

**Comment by eileenmcnaughton** [comment]
@colemanw right - the only entity that does have one is entity_tag I guess. 
The possibilities are

- financial_type
- payment_processor
- option_value.value WHERE option_group_id:name = Payment Instrument

(that last doesn't help!)

**Comment by colemanw** [comment]
More than just entityTag, there's quite a few `entity_table` fields with pseudoconstants. It's pretty much required to get a bridge join to work.
I don't understand that last one - how does that point to a table name?

**Comment by eileenmcnaughton** [comment]
@colemanw here is the latter part of the table

image


The entity_id actually holds the id not the value it seems - which is a nasty non standard use of the option_value table but might be helpful here

**Comment by colemanw** [comment]
Yes that is actually helpful. So our pseudoconstant list just needs to contain 3 options and we should be good to go, at least as far as SearchKit is concerned.

**Comment by eileenmcnaughton** [comment]
@colemanw ok - that seemed to work - let's see how Search kit gets on

**Comment by colemanw** [comment]
I don't think the test failure is related but retest this please just to be sure.

**Comment by eileenmcnaughton** [comment]
@colemanw sadly it is - it's the contribution recur entity defaults tripping us up (AGAIN) - I added this
https://github.com/civicrm/civicrm-core/pull/19934

**Comment by colemanw** [comment]
@eileenmcnaughton I've merged the other. Can you rebase this pls?

**Comment by eileenmcnaughton** [comment]
@colemanw done - lets see how it goes!

**Comment by eileenmcnaughton** [comment]
@colemanw yep passing now!

**Comment by colemanw** [comment]
Looks great!

**Pull request merged**